h2o-py/h2o/model/dim_reduction.py

```diff
@@ -94,20 +93,23 @@ def proj_archetypes(self, test_data, reverse_transform=False):
         return h2o.get_frame(j["model_metrics"][0]["predictions"]["frame_id"]["name"])
 
 
-    def screeplot(self, type="barplot", server=False):
+    def screeplot(self, type="barplot", server=False, save_plot_path=None):
         """
         Produce the scree plot.
 
         Library ``matplotlib`` is required for this function.
 
         :param str type: either ``"barplot"`` or ``"lines"``.
         :param bool server: if true set server settings to matplotlib and do not show the graph
+        :param save_plot_path: a path to save the plot via using mathplotlib function savefig
+        
+        :returns: Object that contains the resulting scree plot (can be accessed like result.figure).
         """
         # check for matplotlib. exit if absent.
         plt = get_matplotlib_pyplot(server)
         if plt is None:
             return
```

Comment:
@koniecsveta If we do not return anything - wouldn't that cause strange errors?

cc: @sebhrusen

Reply:
we could call `get_matplotlib_pyplot(server, raise_if_not_available=True)` in most of those plots I guess.
I don't see the point of just ignoring here if we can't produce anything.

Reply:
I think, if it would, there already would be some. For example  `_plot_pr` already returns "something" but on matplotlib absent it returns anything. But we can add warning on matplotlib absent

Reply:
well, `get_matplotlib_pyplot(server)` will still print a message if `matplotlib` is missing, even when it doesn't raise.
I'm not sure I understand the rationale behind that though: when user asks something and we can't do it, we should raise imo. @tomasfryda?

Reply:
@koniecsveta do you mean that because some of those plot functions return something (eg. p/r) then we still want to return those values even if the plot can't be drawn?
It was a strange original design indeed: for example, there are much better ways to obtain the p/r than calling the plot function… so here, by not raising, it's as if we privilege the secondary contract (return value) over the first one (drawing a plot).

Reply:
the thing to keep in mind - prior to this change the functions didn't have much use in scripts, users could see a warning that something is wrong

with this change we are enabling usage in scripts and we need to think about proper error reporting and what we return to the caller

Reply:
> I'm not sure I understand the rationale behind that though: when user asks something and we can't do it, we should raise imo. @tomasfryda?
@sebhrusen This was the behavior before refactoring it into the function so the rationale was to be backward compatible. If we are ok to break the backward compatibility I'd be in favor of always raising an error.

Reply:
we cannot break backwards compatibility in a fix release cycle

can we instead raise the error when the returned object is accessed?

Reply:
I think we can, when figure will be private and getter will check whether there was saved some raise_error_flag into it

Reply:
b9567fe